src/Draco.Fuzzing/Fuzzer.cs

```diff
@@ -68,17 +70,23 @@ private sealed class QueueEntry(TInput input, ExecutionResult? executionResult =
     /// </summary>
     public required ITracer<TInput> Tracer { get; init; }
 
-    private readonly Queue<QueueEntry> inputQueue = new();
-    private readonly HashSet<TCoverage> seenCoverages = [];
+    /// <summary>
+    /// True if the fuzzer should run in multithreaded mode. Only recommended for out-of-process execution.
+    /// </summary>
+    public bool Multithread { get; init; }
+
+    private readonly BlockingCollection<QueueEntry> inputQueue = [];
+    private readonly ConcurrentHashSet<TCoverage> seenCoverages = [];
+    private readonly object tracerSync = new();
```

Comment:
A reason to bump to .NET 9 rc1: https://learn.microsoft.com/en-us/dotnet/api/system.threading.lock?view=net-9.0

Reply:
I've seen this but we won't bump to .NET 9 before release. Then we can change it